pkg/planner/core/logical_plan_builder.go

```diff
@@ -4446,14 +4447,17 @@ func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName, as
 	if tblName.L == "" {
 		tblName = tn.Name
 	}
-
 	if tableInfo.GetPartitionInfo() != nil {
+		// When table has global index in the partiton, we disable the partition prune mode.
```

Comment:
```suggestion
		// When table has global index in the partiton, we disable the static partition prune mode.
```

Maybe static prune mode?

Reply:
No, The dynamic partition prune mode will be closed.

Reply:
remove this changement